tests/unit/static/test_static.py

```diff
@@ -99,6 +99,13 @@ def test_dependency_tracker_add_resources_wit_deduplication():
     assert set(dep) == {'res1', 'res2'}
 
 
+def test_dependency_tracker_insertion_order():
```

Comment:
Could probably add a test that shows that things in resources that are also in the initial deps aren't duplicated? The test above this one also needs changing I think because it's not actually testing what it says it is (it does the dedup itself in the test!)

Reply:
Fixed the other test case instead

Reply:
This test might pass even when the data type would be a set. (At least the string representation of the set is ordered alphabetically.)
I would suggest to swap lines 104/105 so the result is not in alphabetical order.